src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -524,6 +550,10 @@ def construct_containerapp(self):
                 set_managed_identity(self.cmd, self.get_argument_resource_group_name(), containerapp_def, system_assigned=True)
             else:
                 set_managed_identity(self.cmd, self.get_argument_resource_group_name(), containerapp_def, user_assigned=[self.get_argument_registry_identity()])
+
+        if self.get_argument_source():
+            app = self.set_up_create_containerapp_if_source_or_repo(containerapp_def=containerapp_def)
+            containerapp_def = self.set_up_create_containerapp_source(app=app, containerapp_def=containerapp_def)
```

Comment:
Why do you need two steps to construct container app payload? Is `app` required?

Reply:
app is an instance of [ContainerApp](https://github.com/Greedygre/azure-cli-extensions/blob/xinyu/containerapp_create_refactor/src/containerapp/azext_containerapp/_up_utils.py#L348) and is required in the `set_up_create_containerapp_source` method to properly set up source and repo for containerapp create (similar to containerapp up)